src/screens/Forum/ForumPostScreen.js

```diff
@@ -1,10 +1,127 @@
-import React from 'react';
-import { View, Text } from 'react-native';
+import React, { useState, useEffect, useCallback } from 'react';
+import { View, Text, RefreshControl } from 'react-native';
+import PropTypes from 'prop-types';
+import firestore from '@react-native-firebase/firestore';
+import {
+  Avatar, Button, Card, Paragraph,
+} from 'react-native-paper';
+import { ScrollView } from 'react-native-gesture-handler';
+
+function wait(timeout) {
+  return new Promise((resolve) => {
+    setTimeout(resolve, timeout);
+  });
+}
+
+export default function ForumPostScreen({ navigation }) {
+  const postId = navigation.getParam('postId');
+  const [errorMessage, setErrorMessage] = useState(null);
+  const [post, setPost] = useState(null);
+  const [replies, setReplies] = useState([]);
+  const [refreshing, setRefreshing] = useState(false);
+
+  const getData = useCallback(() => {
+    /* Query forum post data from post id passed in through navigation */
+    firestore().collection('forum_posts').doc(postId)
+      .get()
+      .then((postSnap) => {
+        const postData = postSnap.data();
+
+        /* Query post author user data */
+        const authorSnapshot = firestore().collection('users').doc(postData.userID).get();
+
+        Promise.all([authorSnapshot]).then((authorSnap) => {
+          if (authorSnap.length) {
+            const authorData = authorSnap[0].data();
+
+            /* Set post hook with post and author data */
+            setPost(
+              <Card>
+                <Card.Title title={postData.title} subtitle={`${authorData.displayName} ${postData.createdAt.toDate().toLocaleTimeString('en-US')}`} left={(props) => <Avatar.Text {...props} label={authorData.displayName.charAt(0).toUpperCase()} />} />
+                <Card.Content>
+                  <Paragraph>{postData.body}</Paragraph>
+                </Card.Content>
+                <Card.Actions>
+                  <Button>Reply</Button>
+                </Card.Actions>
+              </Card>,
+            );
+
+            /* Query for post comment data */
+            firestore().collection('forum_comments').where('postID', '==', postId)
+              .get()
+              .then((snapshot) => {
+                const repliesData = snapshot.docs.map((doc) => doc.data());
```

Comment:
(See below) You can make this `const repliesData = snapshot.docs.map((doc) =>({ ...doc.data(), id: doc.id }));` to add the comment's IDs to this array.